csharp/src/Ice/InputStream-Class.cs

```diff
@@ -528,15 +528,15 @@ private bool SkipSlice()
             Debug.Assert(_current != null);
             if (Communicator.TraceLevels.Slicing > 0)
             {
-                ILogger logger = Communicator.Logger;
-                string slicingCat = Communicator.TraceLevels.SlicingCat;
-                if (_current.InstanceType == InstanceType.Exception)
+                string typeId = _current.SliceTypeId ?? "";
```

Comment:
This change does not make sense to me, so I think it should be reverted.

When I get an exception or class instance sliced, I want to see the trace every single time a slice is sliced, not just once per execution of the whole application.

Reply:
I just move the code from TraceUtil to InputStream, but it is doing the same as before

https://github.com/zeroc-ice/ice/blob/ea953a66794c75c92a7619cd56f523be89f03e16/csharp/src/Ice/TraceUtil.cs#L61

We can simplify and do the trace all the time

Reply:
I didn't realize we suppressed all this tracing. And yes, I think we should output a slicing 
trace each time we slice something.

Maybe we could reduce it to once per InputStream, so if you have huge sequence of class-instances that all get sliced, you don't get many traces. I would still find this behavior confusing: it would sound like only my first element is getting sliced while all the other ones are already the correct type? So better keep a trace per actual slicing.

Reply:
Ok I will update it to trace the actual slicing